modules/light-clients/08-wasm/internal/types/store.go

```diff
@@ -0,0 +1,226 @@
+package types
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	wasmvmtypes "github.com/CosmWasm/wasmvm/v2/types"
+
+	"cosmossdk.io/store/cachekv"
+	"cosmossdk.io/store/tracekv"
+	storetypes "cosmossdk.io/store/types"
+)
+
+var (
+	_ wasmvmtypes.KVStore = &StoreAdapter{}
+	_ storetypes.KVStore  = &MigrateClientWrappedStore{}
+
+	SubjectPrefix    = []byte("subject/")
+	SubstitutePrefix = []byte("substitute/")
+)
+
+// MigrateClientWrappedStore combines two KVStores into one.
+//
+// Both stores are used for reads, but only the subjectStore is used for writes. For all operations, the key
+// is checked to determine which types to use and must be prefixed with either "subject/" or "substitute/" accordingly.
+// If the key is not prefixed with either "subject/" or "substitute/", a default action is taken (e.g. no-op for Set/Delete).
+type MigrateClientWrappedStore struct {
```

Comment:
I'm gonna go ahead and suggest that we should actually name this `MergedClientStore`.

I actually disagree with the naming of `MigrateClientWrappedStore`. This is only used in client recovery cases, and the naming is misleading and adds confusion to different workflows such as `MigrateContract` which will likely be used more heavily at some point in the future. 
There is also "migrations" which handle upgrades such as moving stored checksums to a keyset collection. 

I think `MergedClientStore` makes sense given what it's used for and considering it is now `internal` and is well documented it should be fine to change this imo! What do you think?

Reply:
Thank you!! 🙏 

Definitely an overloaded term in this module, for sure.